src/Pos/Util.hs

```diff
@@ -250,12 +252,12 @@ forcePutMVar mvar val = do
 -- satisfies, it is returned.
 readMVarConditional :: (MonadIO m) => (x -> Bool) -> MVar x -> m x
 readMVarConditional predicate mvar = do
-    rData <- liftIO . readMVar $ mvar -- first we try to read for optimization only
+    rData <- readMVar mvar -- first we try to read for optimization only
     if predicate rData then pure rData
     else do
-        tData <- liftIO . takeMVar $ mvar -- now take data
```

Comment:
Sorry, I wasn't aware you're currently working on it. Okay, I can remove these changes.